### PR TITLE
Dejank navigation between thread posts

### PR DESCRIPTION
**Pull request by gaearon**
Currently, when opening a thread post, we'd show two spinners: the one below and the one above. This feels a bit janky/busy and distracts from actually reading the post you clicked on. Twitter solves this by scrolling you down just enough that you *don't see* the parent post (and so there is no need to show a spinner). I'm doing the same behavior here.

This also lets us remove some fragile logic that's bound to eventually get out of sync.

## Before

Web:

https://github.com/bluesky-social/social-app/assets/810438/dec6c9a3-0864-4515-a19f-cd668b556200

iOS:

https://github.com/bluesky-social/social-app/assets/810438/87bc5bac-0e1f-484e-96d4-8d7a97133f21

## After

Web:

https://github.com/bluesky-social/social-app/assets/810438/37b51efb-04c0-4375-a35c-f578689d9ec6

iOS:

https://github.com/bluesky-social/social-app/assets/810438/158463d5-1f29-4c26-9cf9-8ea835d665a4

(Sorry, my Android emulator is still hosed.)

### CODE REVIEW

File: src/view/com/post-thread/PostThread.tsx [hunk 1]
```diff
@@ -507,12 +490,15 @@ function* flattenThreadSkeleton(
...
   if (node.type === 'post') {
-    if (node.parent) {
-      yield* flattenThreadSkeleton(node.parent, hasSession, treeView)
-    } else if (node.ctx.isParentLoading) {
-      yield PARENT_SPINNER
+    if (!node.ctx.isParentLoading) {
+      if (node.parent) {
+        yield* flattenThreadSkeleton(node.parent, hasSession, treeView, false)
+      } else if (!isTraversingReplies) {
+        yield TOP_COMPONENT
```
Comment (gaearon) [inline src/view/com/post-thread/PostThread.tsx:500]: Note this is now rendered only when the parent tree has loaded. This prevents jumps.

File: src/view/screens/PostThread.tsx [hunk 2]
```diff
@@ -79,9 +75,6 @@ export function PostThreadScreen({route}: Props) {
 
   return (
     <View style={s.hContentRegion}>
-      {isMobile && (
-        <ViewHeader title={_(msg({message: 'Post', context: 'description'}))} />
```
Comment (gaearon) [inline src/view/screens/PostThread.tsx:83]: This moved to always be in the flat list instead of sometimes-here-sometimes-there.

**Comment by gaearon** [comment]
Pushed another commit that removes the top border from the highlighted post unless it's a root. It's cleaner (tm).

## Web

https://github.com/bluesky-social/social-app/assets/810438/accd6452-9adf-4968-9c8a-a4b5ed49ff2e

## iOS

https://github.com/bluesky-social/social-app/assets/810438/a61d6ebf-e2a8-469d-b865-08e562d79aec

**Comment by gaearon** [comment]
Resolved conflicts.

**Review by pfrazee — APPROVED** [review]
LGTM!

**Pull request merged**